tiledb/cloud/_results/results.py

```diff
@@ -101,7 +101,7 @@ def _validate(self, attribute, value):
         del attribute, value  # unused
         if self.results_stored and not self.task_id:
             raise ValueError("task_id must be set for stored results")
-        if not self.results_stored and not self._body:
+        if not self.results_stored and self._body is None:
```

Comment:
Interesting this is `None` vs `""` -- I do believe you've tested this correctly. Is it worth adding

```
if not self.results_stored and (self._body is None or self._body == ""):
```

?

(just a thought)

Reply:
We specifically don’t; this needs to test that the `_body` is not provided (results were not downloaded) versus empty (results were downloaded and were zero-length).